core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -41,6 +43,15 @@ abstract class SparkFunSuite
     }
   }
 
+  // helper function
+  protected final def getFile(file: String): File = {
```

Comment:
I like what you've done here with `getCanonicalPath`. Does this call here need `getCanonicalFile` for good measure? I supposes there's also something like `Paths.get(... toURI).toFile` but I don't know if that acts any differently.

How about calling these `getTestResourceFile` and `getTestResourcePath`?